kotlinpoet/src/commonTest/kotlin/com/squareup/kotlinpoet/TypeSpecTest.kt

```diff
@@ -1061,6 +1102,63 @@ class TypeSpecTest {
     )
   }
 
+  @Test fun classImplementsExtendsPrimaryConstructorNoParams() {
+    val taco = ClassName(tacosPackage, "Taco")
+    val food = ClassName("com.squareup.tacos", "Food")
+    val typeSpec = TypeSpec.classBuilder("Taco")
+      .addModifiers(ABSTRACT)
+      .superclass(AbstractSet::class.asClassName().parameterizedBy(food))
+      .addSuperinterface(Serializable::class)
+      .addSuperinterface(Comparable::class.asClassName().parameterizedBy(taco))
+      .primaryConstructor(FunSpec.constructorBuilder().build())
+      .build()
+    assertThat(toString(typeSpec)).isEqualTo(
+      """
+        |package com.squareup.tacos
+        |
+        |import java.io.Serializable
+        |import java.util.AbstractSet
+        |import kotlin.Comparable
+        |
+        |public abstract class Taco() : AbstractSet<Food>(), Serializable, Comparable<Taco>
```

Comment:
Will this currently simply overflow for longer lists? The suggestion in the style guide is to wrap after the `:`, but I believe it'll be quite hard to recognize this case in the code.

<img width="724" alt="image" src="https://github.com/square/kotlinpoet/assets/1186186/987eecb9-b83b-4a07-8a7d-6bc1f20820b2">

Reply:
I want an API like 
```kotlin
val mark = codeWriter.mark()
val originalLine = codeWriter.lineNumber

// Do normal write 
if (codeWriter.lineNumber > originalLine) {
  codeWriter.reset(mark)

  // Do wrapped write
}
```
Would be useful for so many constricts. Probably even deserves a helper function that just takes two lambdas.